src/client/pages/Section/Descriptions/NationalDataDescriptions/DataSources/Columns/hook/useValidationError.ts

```diff
@@ -0,0 +1,31 @@
+import { useMemo } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { Objects } from 'utils/objects'
+
+import { DataSource } from 'meta/assessment'
+
+import { useCanEditCycleData } from 'client/store/user'
+import { validateLinks } from 'client/pages/Section/Descriptions/NationalDataDescriptions/DataSources/Columns/utils'
+
+type Returned = string
+
+type Props = {
+  dataSource: DataSource
+}
+
+export const useValidationError = (props: Props): Returned => {
+  const { dataSource } = props
+  const { t } = useTranslation()
+  const editor = useCanEditCycleData()
+
+  const validationError = useMemo<Returned>(() => {
+    if (!dataSource.placeholder && Objects.isEmpty(dataSource.reference)) return t('generalValidation.notEmpty')
+
+    if (editor && !validateLinks(dataSource.reference)) return t('generalValidation.invalidLink')
```

Comment:
Currently, the editor WYSIWYG component is agnostic of validations and error states, so I decided to add this logic here instead (to be added to other places if we decide to continue this way). Would we want to keep all validations as a responsibility of the caller components or should we introduce validation logic to the editor itself?

Reply:
Sure, I keep that in mind. We will keep this validation as responsibility of the EditorWYSIWYG component callers then. 
Another option could be to do something like:
```
<EditorWYSIWYG
     disabled={disabled}
     onChange={onChange}
     ...
     validators={[EditorValidators.links, EditorValidators.otherValidator]}
     onValidationErrors={setErrors}
 >
 ```
 There are pros and cons to both approaches. What do you think?  
 cc @sorja

Reply:
Since all we need to do now is show a tooltip, I agree. If we needed to do something else in case of error, I would say better keep it to the caller, but that's not the case. 👍